src/lib/Libifl/int_rdrpy.c

```diff
@@ -83,18 +84,22 @@ PBSD_rdrpy_sock(int sock, int *rc)
 		return NULL;
 	}
 
-	DIS_tcp_funcs();
-	old_timeout = pbs_tcp_timeout;
-	if (pbs_tcp_timeout < PBS_DIS_TCP_TIMEOUT_LONG)
-		pbs_tcp_timeout = PBS_DIS_TCP_TIMEOUT_LONG;
+	if (prot == PROT_TCP) {
+		DIS_tcp_funcs();
+		old_timeout = pbs_tcp_timeout;
+		if (pbs_tcp_timeout < PBS_DIS_TCP_TIMEOUT_LONG)
+			pbs_tcp_timeout = PBS_DIS_TCP_TIMEOUT_LONG;
+	}
 
-	if ((*rc = decode_DIS_replyCmd(sock, reply)) != 0) {
+	if ((*rc = decode_DIS_replyCmd(sock, reply, prot)) != 0) {
 		(void)free(reply);
 		pbs_errno = PBSE_PROTOCOL;
 		return NULL;
 	}
-	dis_reset_buf(sock, DIS_READ_BUF);
-	pbs_tcp_timeout = old_timeout;
+	if (prot == PROT_TCP) {
+		dis_reset_buf(sock, DIS_READ_BUF);
```

Comment:
also why would dis_reset_buf() not be required for TPP? It clears of the read buffer from DIS layer in case something was pending - perhaps best to to leave it on for both protocols?